File: 67-add-binary/add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res = "";
        int carry = 0;
        int i = a.length()-1;
        int j = b.length()-1;
        while(i>=0 || j>=0 || carry) {
            if(i >= 0) {
                carry += a[i] - 48;
                i--;
            }
            if(j >= 0) {
                carry += b[j] - 48;
                j--;
            }
            res += carry%2 + 48;
            carry /= 2;
        }
        reverse(res.begin(),res.end());
        return res;
    }
};

File: 67-add-binary/add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        carry = 0
        res = ""
        i = len(a) - 1
        j = len(b) - 1
        while i >= 0 or j >= 0 or carry :
            if i >= 0 :
                carry += int(a[i])
                i -= 1
            if j >= 0 :
                carry += int(b[j])
                j -= 1
            res += str(carry % 2)
            carry //= 2
        return res[::-1]            